lwatools/file_tools/handle_records.py

```diff
@@ -4,127 +4,171 @@
         self.loaded_filename = recordset_fname
 
-    def save(self, record_set_fname:str)->None:
-        self.df.to_json(path_or_buf=record_set_fname)
-        self.saved_filename(record_set_fname)
-
-    # def add_base_paths(self, paths:Union[List[str], str])->None:
-    #     #TODO this won't be used I don't think
-    #     if type(paths) == str:
-    #         # ensure path ends with slash
-    #         if paths.endswith('/'):
-    #             paths+='/'
-    #         self.file_paths.append(paths)
-    #     elif type(paths) == list:
-    #         # ensure path ends with slash
-    #         for path in paths:
-    #             if path.endswith('/'):
-    #                 path+='/'
-    #             self.file_paths.append(path)
+    def save(self, record_set_fname:Optional[str]=None)->None:
+        if record_set_fname is not None:
+            self.df.to_csv(path_or_buf=record_set_fname, columns=COLUMNS, sep=',', header=COLUMNS)
+            self.saved_filename = record_set_fname
+            print(f'Wrote recordset to {record_set_fname}')
+        else:
+            try:
+                self.df.to_csv(path_or_buf=self.loaded_filename, columns=COLUMNS, sep=',', header=COLUMNS)
+                self.saved_filename = self.loaded_filename
+                print(f'Wrote recordset to {self.loaded_filename}')
+            except ValueError:
+                raise ValueError ('Need an argument for output filename')
+            except FileNotFoundError:
+                raise FileNotFoundError ("loaded_filename somehow doesn't exist. Please pass an argument for output filename.")
 
     def add_observation(self,
         tbn_path:str, sdf_path:str, tx_type:Literal['tone', 'morse', 'sweep',
         'other'], station: Literal['lwasv', 'lwa1'], known_transmitter:Optional[str]=None,
-        notes: Optional[str]=None, tx_fc:Optional[float]=None, corrupted:bool=False):
+        notes: Optional[str]=None, tx_fc:Optional[float]=None, corrupted:bool=False,
+        tarball_path:Optional[str]=None):
         """
         Adds an observation to the dataframe - requires that a dataframe has been loaded previously
         """
 
-        #TODO wait does this work. How do we initially add observations.
-        assert not self.df.empty, "Must load a record set file first!"
 
-        if known_transmitter:
-            if known_transmitter not in known_transmitter_locations.keys():
-                raise ValueError(f"Unrecognized transmitter name {known_transmitter}. See utils/known_transmitters.py.")
+        if tx_type == 'tone' and known_transmitter not in known_transmitter_locations.keys():
+            raise ValueError(f"Unrecognized transmitter name {known_transmitter}. See utils/known_transmitters.py.")
 
         obs_dict = defaultdict() # handles keyerrors
         
+        if tbn_path.startswith('./'):
+            raise ValueError("Relative paths are not allowed")
```

Comment:
Why not? Can't we just use `Path.resolve()` to expand?

Reply:
Probably yea, I'll check